src/pcp/pidstat/pcp-pidstat.py

```diff
@@ -360,11 +362,17 @@ def print_report(self, timestamp, ncpu):
 
             if self.pidstat_options.per_processor_usage and total_percent is not None:
                 total_percent = float("%.2f"%(total_percent/ncpu))
-
-            if self.pidstat_options.show_process_user:
-                self.printer("%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s" % (timestamp,process.user_name(),process.pid(),user_percent,system_percent,guest_percent,total_percent,process.cpu_number(),process.process_name()))
+            if PidstatOptions.timefmt != None:
+                tstamp = datetime.fromtimestamp(time.time()).strftime(PidstatOptions.timefmt)
```

Comment:
When we're re factoring this to a single check (around line 575), we should take care in using the group.timestamp so that the time in the output is accurate to when the metrics were actually gathered (not displayed).